src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,40 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  health = 100;
+  constructor(name) {
+    this.name = name;
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hidden = false;
+  constructor(name) {
+    super(name);
+    Animal.alive.push(this);
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    Animal.alive.push(this);
+  }
+
+  bite(herbivore) {
+    if (herbivore.hidden === false && !(herbivore instanceof Carnivore)) {
+      herbivore.health -= 50;
+    }
```

Comment:
The condition in the `bite` method should only check if the `herbivore` is an instance of `Herbivore` and is not hidden. Currently, it incorrectly checks if the `herbivore` is not an instance of `Carnivore`, which is not the same as checking for an instance of `Herbivore`.